components/brave_wallet/browser/keyring_service.cc

```diff
@@ -2009,6 +2009,21 @@ std::optional<std::vector<uint8_t>> KeyringService::GetZCashPubKey(
   return zcash_keyring->GetPubkey(*key_id);
 }
 
+std::optional<std::array<uint8_t, kOrchardRawBytesSize>>
+KeyringService::GetOrchardRawBytes(const mojom::AccountIdPtr& account_id,
```

Comment:
this should be unit tested

Reply:
We have GetOrchardRawBytes in ZCashKeyringUnitTests

Reply:
I'm not following, what does that have to do with this method? There is logic in `KeyringService::GetOrchardRawBytes` that needs to be tested to ensure if works correctly. You should have test cases that cover when the keyring can't be found and when it can be found. Also it seems like we should using `base::expected` here so we can return better error info for missing keyring vs a problem with GetOrchardRawBytes? We're moving the error handling to here https://github.com/brave/brave-core/pull/23076/files#diff-5b8bf71df4ea42bbeb85a5430e901d57fce62becb1c8212e27ce97aa23441e7fR121 where we don't know what happened so we just produce a generic internal error

Reply:
you can discuss internally for wallet whether to change to `base::expected` in this PR or convert all similar methods in a separate PR, but we definitely need tests for new functionality that is added here